staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -280,15 +276,15 @@ func (c *Controller) syncHandler(ctx context.Context, key string) error {
 		// We choose to absorb the error here as the worker would requeue the
 		// resource otherwise. Instead, the next time the resource is updated
 		// the resource will be queued again.
-		utilruntime.HandleError(fmt.Errorf("%s: deployment name must be specified", key))
+		utilruntime.HandleError(fmt.Errorf("%#v: deployment name must be specified", objectRef))
 		return nil
 	}
 
 	// Get the deployment with the name specified in Foo.spec
 	deployment, err := c.deploymentsLister.Deployments(foo.Namespace).Get(deploymentName)
 	// If the resource doesn't exist, we'll create it
 	if errors.IsNotFound(err) {
-		deployment, err = c.kubeclientset.AppsV1().Deployments(foo.Namespace).Create(context.TODO(), newDeployment(foo), metav1.CreateOptions{})
+		deployment, err = c.kubeclientset.AppsV1().Deployments(foo.Namespace).Create(context.TODO(), newDeployment(foo), metav1.CreateOptions{FieldManager: FieldManager})
```

Comment:
Adding a field manager to non-apply requests is a little unusual.  What does this do for future us?

Reply:
I thought it was normal to supply the FieldManager in all requests. I see that it is in all the XXXOptions types, and the apiserver writes into the ManagedFields in all write operations, so those are consistent with supplying FieldManger in all writes.

Reply:
@deads2k is this ok to leave in?

Reply:
Seems to be correct to pass the field manager here too. ManagedFields is always written.